### PR TITLE
cephadm: Refactor, prepare for other adoptions

### PR DESCRIPTION
**Pull request by swiftgist**
Move the prometheus adopt out of command_adopt. refactor skip pull,
legacy check and fsid lock.

Signed-off-by: Eric Jackson <swiftgist@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by swiftgist** [comment]
an fyi, the `command_adopt` is essentially the same, but moving the legacy check unindented the entire function.

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -2404,157 +2404,162 @@ def command_adopt():
...
+    if daemon_type in Ceph.daemons:
+        command_adopt_ceph(daemon_type, daemon_id, fsid);
+    elif daemon_type == 'prometheus':
+        command_adopt_prometheus(daemon_id, fsid)
+    elif daemon_type == 'grafana':
+        raise Error('adoption of grafana not implemented')
+    elif daemon_type == 'node-exporter':
+        raise Error('adoption of node-exporter not implemented')
+    elif daemon_type == 'alertmanager':
+        raise Error('adoption of alert-manager not implemented')
```
Comment (mgfritch) [inline src/cephadm/cephadm]: seems like this is the same as the default `else` block?

File: src/cephadm/cephadm [hunk 2]
```diff
@@ -2404,157 +2404,162 @@ def command_adopt():
...
+    else:
+        raise Error('daemon type %s not recognized' % daemon_type)
```
Comment (mgfritch) [inline src/cephadm/cephadm]: Maybe just handle anything not implemented in a generic way:
```suggestion
        raise Error('adoption of daemon type %s is not implemented' % daemon_type)
```

File: src/cephadm/cephadm (hunk 1 above)
Comment (swiftgist) [reply src/cephadm/cephadm]: Figured I would be explicit about what's next to be implemented.... although I could see this collapsing into a `Monitoring.components` check

File: src/cephadm/cephadm (hunk 2 above)
Comment (swiftgist) [reply src/cephadm/cephadm]: I haven't gone through grafana, node-exporter and alertmanager, so I'm not quite sure whether we will have three separate functions or if the current command_adopt_prometheus will become a command_adopt_monitoring.

**Pull request merged**